ts/src/pro/bitmart.ts

```diff
@@ -1331,9 +1362,40 @@ export default class bitmart extends bitmartRest {
             orderbook['datetime'] = this.iso8601 (timestamp);
             const messageHash = table;
             client.resolve (orderbook, messageHash);
+            // resolve ForSymbols
+            const messageHashForMulti = channel + ':' + symbol;
+            client.resolve (orderbook, messageHashForMulti);
         }
     }
 
+    async watchOrderBookForSymbols (symbols: string[], limit: Int = undefined, params = {}): Promise<OrderBook> {
+        /**
+         * @method
+         * @name bitmart#watchOrderBookForSymbols
+         * @description watches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @see https://developer-pro.bitmart.com/en/spot/#public-depth-increase-channel
+         * @param {string[]} symbols unified array of symbols
+         * @param {int} [limit] the maximum amount of order book entries to return
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/#/?id=order-book-structure} indexed by market symbols
+         */
+        await this.loadMarkets ();
+        symbols = this.marketSymbols (symbols, undefined, false, true);
+        if (symbols.length > 20) {
+            throw new NotSupported (this.id + ' watchOrderBookForSymbols() accepts a maximum of 20 symbols in one request');
+        }
+        const market = this.market (symbols[0]);
+        let channel = undefined;
+        [ channel, params ] = this.handleOptionAndParams (params, 'watchOrderBookForSymbols', 'depth', 'depth/increase100');
+        let type = 'spot';
+        [ type, params ] = this.handleMarketTypeAndParams ('watchOrderBookForSymbols', market, params);
+        if (type === 'swap' && channel === 'depth/increase100') {
```

Comment:
isn't depth/increase100 compatible with swap?

Reply:
no, it's not compatible: https://developer-pro.bitmart.com/en/futures/#public-depth-channel:~:text=Level%20Depth%20Channel-,futures/depth50,50%20Level%20Depth%20Channel,-Response